### ISSUE TITLE
Geometry not loading

### ISSUE THREAD

**Issue by nabiljebar** [opensourceBIM/BIMserver#633]
Hi,

the geometry is not loading after clicking the eye icon

i have this message

Model data successfully loaded
in JS console i have preloadModel 65539: 372.49609375ms

but nothing else 

Relaease 1.5.88 war version
CentOS release 6.5 (Final)
tomcat 8

**Comment by rubendel** [comment]
Have you had a look at this issue?

https://github.com/opensourceBIM/BIMserver/issues/630

**Comment by nabiljebar** [comment]
the default render engine is  "Ifc Open Shell"

I  re-installed BIMsurfer and it is not working after a  new check in

**Comment by rubendel** [comment]
Nothing in the server log too?

**Comment by nabiljebar** [comment]
absolutely nothing

**Comment by rubendel** [comment]
That is also strange, can you paste the full log?

**Comment by nabiljebar** [comment]
sorry but in bimserver.log i have this

2017-10-20 15:46:56,685 INFO [pool-42-thread-1] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:627] Writing debug file to /home/opt/tomcat8/webapps/bimserver/WEB-INF/debug/2017-10-20-15-46-55/IfcSlab-error-2.ifc
2017-10-20 15:46:56,900 ERROR [pool-42-thread-4] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:581] 
org.bimserver.plugins.renderengine.RenderEngineException: java.io.EOFException
	at org.ifcopenshell.IfcGeomServerClient.<init>(IfcGeomServerClient.java:70)
	at org.ifcopenshell.IfcOpenShellEngine.init(IfcOpenShellEngine.java:69)
	at org.bimserver.renderengine.NoPoolingRenderEnginePool.borrowObject(NoPoolingRenderEnginePool.java:34)
	at org.bimserver.StreamingGeometryGenerator$Runner.run(StreamingGeometryGenerator.java:232)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException: null
	at com.google.common.io.LittleEndianDataInputStream.readAndCheckByte(LittleEndianDataInputStream.java:226)
	at com.google.common.io.LittleEndianDataInputStream.readInt(LittleEndianDataInputStream.java:116)
	at org.ifcopenshell.IfcGeomServerClient.<init>(IfcGeomServerClient.java:56)
	... 8 common frames omitted
2017-10-20 15:46:56,901 INFO [pool-42-thread-4] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:627] Writing debug file to /home/opt/tomcat8/webapps/bimserver/WEB-INF/debug/2017-10-20-15-46-55/IfcBeam-error-4.ifc
2017-10-20 15:46:56,959 ERROR [pool-42-thread-3] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:581] 
org.bimserver.plugins.renderengine.RenderEngineException: java.io.EOFException
	at org.ifcopenshell.IfcGeomServerClient.<init>(IfcGeomServerClient.java:70)
	at org.ifcopenshell.IfcOpenShellEngine.init(IfcOpenShellEngine.java:69)
	at org.bimserver.renderengine.NoPoolingRenderEnginePool.borrowObject(NoPoolingRenderEnginePool.java:34)
	at org.bimserver.StreamingGeometryGenerator$Runner.run(StreamingGeometryGenerator.java:232)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException: null
	at com.google.common.io.LittleEndianDataInputStream.readAndCheckByte(LittleEndianDataInputStream.java:226)
	at com.google.common.io.LittleEndianDataInputStream.readInt(LittleEndianDataInputStream.java:116)
	at org.ifcopenshell.IfcGeomServerClient.<init>(IfcGeomServerClient.java:56)
	... 8 common frames omitted
2017-10-20 15:46:56,960 INFO [pool-42-thread-3] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:627] Writing debug file to /home/opt/tomcat8/webapps/bimserver/WEB-INF/debug/2017-10-20-15-46-55/IfcColumn-error-0.ifc
2017-10-20 15:46:56,962 INFO [LongStreamingCheckinAction] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:960] Rendertime: 1161 msms, Reused (by hash): 0 B, Reused (by transformation): 0 B, Reused (by mapping): 0 B, Total: 0 B, Final: 0 B
2017-10-20 15:46:56,962 INFO [LongStreamingCheckinAction] o.b.StreamingGeometryGenerator [StreamingGeometryGenerator.java:961] Saved color data: 0 B
2017-10-20 15:47:01,068 INFO [http-nio-8080-exec-6] o.b.c.NewDiskCacheManager [NewDiskCacheManager.java:117] Start caching b7c546c0126cba9f62b4d0a5063c1dc0
2017-10-20 15:47:01,382 INFO [http-nio-8080-exec-6] o.b.c.NewDiskCacheOutputStream [NewDiskCacheOutputStream.java:59] Renaming temp file b7c546c0126cba9f62b4d0a5063c1dc0.__tmp to b7c546c0126cba9f62b4d0a5063c1dc0
2017-10-20 15:47:01,383 INFO [http-nio-8080-exec-6] o.b.c.NewDiskCacheManager [NewDiskCacheManager.java:150] Done caching b7c546c0126cba9f62b4d0a5063c1dc0

**Comment by rubendel** [comment]
Hmm that sounds like an error in IfcOpenShell. Maybe it has to do with (native) dependencies, but you never know with C++ code, there are no errors/warnings. You are using the latest (default) version I assume?

**Comment by nabiljebar** [comment]
tried with 0.5.24, 05.22 and 0.5.22 version of IfcOpenShell
also with ifcengine.JvmRenderEnginePlugin same errors

**Comment by garywangcn** [comment]
I'm also encountering this problem, there is nothing in 3D view:
![image](https://user-images.githubusercontent.com/22828078/31928435-ba6cf8d2-b8ca-11e7-81f2-31f8378c181e.png)


**Below is the console log, don't have any error:**
14:46:39 INFO  [main]: Server started successfully                                                      (JarBimServer.java:143) 
14:48:20 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache 3688127be5f968962380e510005ae6d2                              (NewDiskCacheManager.java:105) 
14:48:20 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache e82dd7c0439ed24c98cf70d8ad144ce0                              (NewDiskCacheManager.java:105) 
14:48:20 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache 42b4a50dabba818d315f47f5c0f303bb                              (NewDiskCacheManager.java:105) 
14:48:20 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache 7aa9cd50683290356b51059da59a2ffd                              (NewDiskCacheManager.java:105) 
14:48:20 INFO  [qtp986944742-32]: {"request":{"interface":"org.bimserver.ServiceInterface","method":"getExtendedDataSchemaByName","parameters":{"name":"VIS_3D_JSON_1_0"}},"token":"2590d8c67a0bfb6a5033749acb6e92a33b7d8f3d2585ae5cdd6c1608d78f1c915f20911a1c789e3253012c7656569bd8"} (JsonHandler.java:69) 
14:48:38 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache 5245a51bc72d3f5197b6903a31b6676e                              (NewDiskCacheManager.java:105) 
14:48:38 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache c47822913d2af15b49cace95a16b51eb                              (NewDiskCacheManager.java:105) 
14:48:38 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache b0361da0dd5c253357a26e75b0416da9                              (NewDiskCacheManager.java:105) 
14:48:38 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache cb9d66b7a1774eb96c95bfe2e3f8feca                              (NewDiskCacheManager.java:105) 
14:48:38 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache b34a8116cf30822802f80b21cb1b4a5e                              (NewDiskCacheManager.java:105) 
14:48:38 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache c54cc3b1130d47f47187824e5f19f57f                              (NewDiskCacheManager.java:105) 
14:48:40 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache a4c62c58a4addfc0a57352eeb7956452                              (NewDiskCacheManager.java:105) 
14:48:40 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache dac3a456baeafe6222dff4e065bad459                              (NewDiskCacheManager.java:105) 
14:48:40 INFO  [org.bimserver.ServiceInterface.download]: Reading from cache 9eae6e62f67d26c46fb28dee209247c0                              (NewDiskCacheManager.java:105)

**Comment by cnyzgkn** [comment]
I met similar problem. Loading model successfully but no rendering

**Comment by zhutianrui** [comment]
I met the same problem and finally I found the reason.

When IfcOpenShell create the Geometry, it needs to run an executable program which will be fail on CentOS 6.5 default install.  You just need to update your gcc to a higher version which supports GLIBCXX_3.4.15 and redirect the libstdc++.so.6  symbolic link to the newest library file. 

I updated the gcc to gcc-5.2.0 and the library file is libstdc++.so.6.0.21. Then model shows up.
